core/src/main/java/hudson/model/Slave.java

```diff
@@ -348,9 +349,14 @@ public FilePath getRootPath() {
     /**
      * Web-bound object used to serve jar files for JNLP.
      */
-    public static final class JnlpJar implements HttpResponse {
+    public static class JnlpJar implements HttpResponse {
```

Comment:
I am not sure that use case is a good idea, but even if it is, you do not need to do this. The JNLP file produced by the plugin can simply point to a custom endpoint which serves a different JAR.